src/compiler/rust/canister/src/lib.rs

```diff
@@ -285,3 +270,527 @@ pub extern "C" fn get_wasm_data() -> WasmData {
 
     serde_json::from_str(std::str::from_utf8(&wasm_data_vec).unwrap()).unwrap()
 }
+
+#[no_mangle]
+#[allow(unused)]
+pub fn init(function_index: i32, pass_arg_data: i32) {
```

Comment:
I like bringing this stuff out of the macros, that's going to make it a lot easier to work on, and it makes a lot of sense with this change. I would have liked to see it come over with a similar structure though

Reply:
Yes I just took the easy path. But very soon, possibly today, you'll be refactoring this again and I hope simplifying it a lot, there's a chance at that. So it's structure might drastically change again